src/main/java/uk/nhs/digital/nhsconnect/lab/results/translator/mapper/PathologyRecordMapper.java

```diff
@@ -16,23 +15,29 @@
     private final PatientMapper patientMapper;
     private final OrganizationMapper organizationMapper;
     private final SpecimenMapper specimenMapper;
+    private final DiagnosticReportMapper diagnosticReportMapper;
     private final ObservationMapper observationMapper;
 
     public PathologyRecord mapToPathologyRecord(final Message message) {
         final PathologyRecordBuilder pathologyRecordBuilder = PathologyRecord.builder();
 
         final var patient = patientMapper.mapToPatient(message);
+        final var specimens = specimenMapper.mapToSpecimens(message, patient);
+        final var performingPractitioner = practitionerMapper.mapToPerformingPractitioner(message);
+        final var performingOrganization = organizationMapper.mapToPerformingOrganization(message);
+        final var observations = observationMapper.mapToTestGroupsAndResults(message);
+        final var diagnosticReport = diagnosticReportMapper.mapToDiagnosticReport(message, patient, specimens,
+            observations, performingPractitioner.orElse(null), performingOrganization.orElse(null));
```

Comment:
If we're going to do all the mappings first into local vars and then populate the record builder, perhaps we can make that separation clear and only create the record builder just before we need it

Reply:
I've split out the mappings into local vars and the populating of the record builder in f181d3d

Let me know what you think

Reply:
Looks good